### PR TITLE
Implement Google Data Catalog as under meta service

### PR DESCRIPTION
**Pull request by jenoudet**
Tackles #12751 
Draft PR

### CODE REVIEW

**Review by yuzhu — COMMENTED** [review]
@jenoudet thanks for the improvement. 
Main high level comment is that Partition and Layout structures both need to be filled out for queries over the table to run. You might want to consult how Glue does it in GlueDatabase.java.  In alluxio catalog land, a non-partitioned table is represented with a single partition. so in that way, every table needs to have its partitionInfo filled out.

File: table/server/underdb/gdc/src/main/java/alluxio/table/under/gdc/GDCDatabase.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+public class GDCDatabase implements UnderDatabase {
+  private static final Logger LOG = LoggerFactory.getLogger(GDCDatabase.class);
+
+  private final UdbContext mUdbContext;
+  private final UdbConfiguration mConfiguration;
+  private final String mGdcDatasetName;
+  private final String mGdcProjectId;
+
+  /* placeholder values */
+  private final String mOwnerName = "PUBLIC_OWNER";
```
[the remark this answers is not in the document]
Comment (jenoudet) [reply table/server/underdb/gdc/src/main/java/alluxio/table/under/gdc/GDCDatabase.java:56]: fixed

File: table/server/underdb/gdc/pom.xml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                        <goals>
+                            <goal>test-jar</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>
```
[the remark this answers is not in the document]
Comment (jenoudet) [reply table/server/underdb/gdc/pom.xml:118]: fixed

File: core/common/pom.xml [hunk 3]
```diff
@@ -31,6 +31,10 @@
 
   <dependencies>
     <!-- External dependencies -->
+<!--    <dependency>-->
```
[the remark this answers is not in the document]
Comment (jenoudet) [reply core/common/pom.xml:34]: It was redundant with the same import somewhere else in the file, I've removed it

**Pull request closed without merging**